linesearch.go

```diff
@@ -149,31 +129,31 @@ func (ls *LinesearchMethod) error(err error) (Operation, error) {
 }
 
 // initNextLinesearch initializes the next linesearch using the previous
-// complete location stored in ls.loc. It fills xNext and returns an evaluation
-// to be performed at xNext.
-func (ls *LinesearchMethod) initNextLinesearch(xNext []float64) (Operation, error) {
-	copy(ls.x, ls.loc.X)
+// complete location stored in loc. It fills loc.X and returns an evaluation
+// to be performed at loc.X.
+func (ls *LinesearchMethod) initNextLinesearch(loc *Location) (Operation, error) {
+	copy(ls.x, loc.X)
```

Comment:
Do we need to set ls.lastOp to be NoOperation up here so that's what it equals if an error occurs?